storm-core/src/jvm/org/apache/storm/cluster/ClusterUtils.java

```diff
@@ -0,0 +1,261 @@
+        for (ExecutorInfo executor : executors) {
+            if (executorStatsMap.containsKey(executor)) {
+                int time = workerHeartbeat.get_time_secs();
+                int uptime = workerHeartbeat.get_uptime_secs();
+                ExecutorStats executorStats = workerHeartbeat.get_executor_stats().get(executor);
+                ExecutorBeat executorBeat = new ExecutorBeat(time, uptime, executorStats);
+                executorWhb.put(executor, executorBeat);
+            }
+        }
+        return executorWhb;
+    }
+
+    public IStormClusterState mkStormClusterStateImpl(Object stateStorage, List<ACL> acls, ClusterStateContext context) throws Exception {
+        if (stateStorage instanceof IStateStorage) {
+            return new StormClusterStateImpl((IStateStorage) stateStorage, acls, context, false);
+        } else {
+            IStateStorage Storage = _instance.mkStateStorageImpl((Map) stateStorage, (Map) stateStorage, acls, context);
+            return new StormClusterStateImpl(Storage, acls, context, true);
+        }
+
+    }
+
+    public IStateStorage mkStateStorageImpl(Map config, Map auth_conf, List<ACL> acls, ClusterStateContext context) throws Exception {
+        String className = null;
+        IStateStorage stateStorage = null;
+        if (config.get(Config.STORM_CLUSTER_STATE_STORE) != null) {
+            className = (String) config.get(Config.STORM_CLUSTER_STATE_STORE);
+        } else {
+            className = "org.apache.storm.cluster.ZKStateStorageFactory";
+        }
+        Class clazz = Class.forName(className);
+        StateStorageFactory storageFactory = (StateStorageFactory) clazz.newInstance();
+        stateStorage = storageFactory.mkStore(config, auth_conf, acls, context);
+        return stateStorage;
+    }
+
+    public static IStateStorage mkStateStorage(Map config, Map auth_conf, List<ACL> acls, ClusterStateContext context) throws Exception {
+        return _instance.mkStateStorageImpl(config, auth_conf, acls, context);
+    }
+
+    public static IStormClusterState mkStormClusterState(Object StateStorage, List<ACL> acls, ClusterStateContext context) throws Exception {
+        return _instance.mkStormClusterStateImpl(StateStorage, acls, context);
+    }
+
+    public static String stringifyError(Throwable error) {
+        String errorString = null;
+        StringWriter result = null;
+        PrintWriter printWriter = null;
+        try {
+            result = new StringWriter();
+            printWriter = new PrintWriter(result);
+            error.printStackTrace(printWriter);
+            if (result != null) {
+                errorString = result.toString();
+            }
+        } finally {
+            try {
+                if (result != null)
+                    result.close();
+                if (printWriter != null)
+                    printWriter.close();
+            } catch (Exception e) {
+            }
+        }
+        return errorString;
+    }
```

Comment:
This can be made a lot simpler with the java 7 AutoClosable semantics.

```
    public static String stringifyError(Throwable error) {
        try (
            StringWriter result = new StringWriter();;
            PrintWriter printWriter = new PrintWriter(result);
        ){
            error.printStackTrace(printWriter);
            return result.toString();
        }
    }
```